### PR TITLE
1.2.0

### PR DESCRIPTION
**Pull request by meatball133**
**Do Not Merge**, This is an in-development pr

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-22.04
+    container:
+      image: crystallang/crystal
+    steps:
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608
+      - name: Build docs
+        run: crystal docs --project-name Representer --project-version 1.2.0-Dev
+      - name: Deploy docs
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        uses: JamesIves/github-pages-deploy-action@v4
```
Comment (ErikSchierboom) [inline .github/workflows/docs.yml:23]: Could you pin this to a commit SHA?

```suggestion
        uses: JamesIves/github-pages-deploy-action@v4
```

File: bin/run.sh [hunk 2]
```diff
@@ -24,17 +24,13 @@ fi
...
-mapping_file="${output_dir}/mapping.json"
-
+solution_files="$(jq -c '.files.solution' ${input_dir}/.meta/config.json)"
 # Create the output directory if it doesn't exist
 mkdir -p "${output_dir}"
 
 echo "${slug}: creating representation..."
 
-./bin/representer "${input_dir}" "${meta_config_json_file}" "${representation_file}" "${mapping_file}" "${representation_config}"
+./bin/representer -s $solution_files -d "${input_dir}" -o "${output_dir}"
```
Comment (ErikSchierboom) [inline bin/run.sh:33]:
```suggestion
./bin/representer -s "${solution_files}" -d "${input_dir}" -o "${output_dir}"
```

Side note: in theory, people could also submit additional files. Does the representer also check those?

**Review by ErikSchierboom — APPROVED** [review]

File: bin/run.sh (hunk 2 above)
Comment (meatball133) [reply bin/run.sh:33]: It does check all files in the config.json file

File: .github/workflows/docs.yml (hunk 1 above)
Comment (meatball133) [reply .github/workflows/docs.yml:23]: I dont understand what you mean

File: bin/run.sh (hunk 2 above)
Comment (ErikSchierboom) [reply bin/run.sh:33]: Okay, but people could also create files themselves and submit them

File: .github/workflows/docs.yml (hunk 1 above)
Comment (ErikSchierboom) [reply .github/workflows/docs.yml:23]: Don't use `@v4` but use `@<commit sha>`. See https://exercism.org/docs/building/github/gha-best-practices#h-pin-actions-to-shas

**Review by ErikSchierboom — APPROVED** [review]

**Review by ErikSchierboom — APPROVED** [review]

**Pull request merged**